pkg/storage/redis/client.go

```diff
@@ -30,7 +61,17 @@ func newClientFromURL(rawurl string, tlsConfig *tls.Config) (*redis.Client, erro
 		}
 		return redis.NewClient(opts), nil
 
-	case "redis-sentinel", "rediss-sentinel", "redis-sentinels":
+	case clusterSchemes.Has(u.Scheme):
+		opts, err := ParseClusterURL(rawurl)
+		if err != nil {
+			return nil, err
+		}
+		if opts.TLSConfig != nil {
+			opts.TLSConfig = tlsConfig
```

Comment:
should we maybe complain when i.e. `redis` scheme is specified but TLS config is not null?

Reply:
The TLS config is optional. by default it would only validate the server name. I guess we could enforce requiring it, but that may break backwards compatibility.